darts/models/forecasting/multivariate_forecasting_model_wrapper.py

```diff
@@ -0,0 +1,140 @@
+"""
+Multivariate forecasting model wrapper
+-------------------------
+
+A wrapper around local forecasting models which allows forecasting multiple components
+"""
+from typing import List, Optional, Tuple
+
+from darts.logging import get_logger
+from darts.models.forecasting.forecasting_model import (
+    FutureCovariatesLocalForecastingModel,
+    LocalForecastingModel,
+    TransferableFutureCovariatesLocalForecastingModel,
+)
+from darts.timeseries import TimeSeries, concatenate, split_multivariate
+from darts.utils.utils import seq2series
+
+logger = get_logger(__name__)
+
+
+class MultivariateForecastingModelWrapper(FutureCovariatesLocalForecastingModel):
+    def __init__(self, model: LocalForecastingModel):
+        """
+        Wrapper around LocalForecastingModel allowing it to predict multivariate TimeSeries.
+
+        A copy of the provided model will be trained on each of the components of the provided series separately.
+        Parameters
+        ----------
+        model
+            Model used to predict individual components
+        """
+        super().__init__()
+
+        self.model: LocalForecastingModel = model
+        self._trained_models: List[LocalForecastingModel] = []
+
+    def _fit(self, series: TimeSeries, future_covariates: Optional[TimeSeries] = None):
+        super()._fit(series, future_covariates)
+        self._trained_models = []
+
+        series = seq2series(series)
+        components = split_multivariate(series)
+        if self.supports_future_covariates:
+            self._trained_models = [
+                self.model.untrained_model().fit(c, future_covariates)
+                for c in components
+            ]
+        else:
+            self._trained_models = [
+                self.model.untrained_model().fit(c) for c in components
+            ]
+
+        return self
+
+    def predict(
+        self,
+        n: int,
+        series: Optional[TimeSeries] = None,
+        future_covariates: Optional[TimeSeries] = None,
+        num_samples: int = 1,
+        **kwargs,
+    ) -> TimeSeries:
+        # we override `predict()` to pass a non-None `series`, so that historic_future_covariates
+        # will be passed to `_predict()` (some future covariates local models require it ex. Kalman)
+        return self._predict(n, future_covariates, num_samples, **kwargs)
+
+    def _predict(
+        self,
+        n: int,
+        future_covariates: Optional[TimeSeries] = None,
+        num_samples: int = 1,
+        verbose: bool = False,
+        **kwargs,
+    ) -> TimeSeries:
+        predictions = [
+            model.predict(n=n, future_covariates=future_covariates)
+            if isinstance(model, FutureCovariatesLocalForecastingModel)
+            else model.predict(n=n)
+            for model in self._trained_models
```

Comment:
Instead of the `isinstance`, rather use the `supports_future_covariates`.

Have a single call the predict with an inline if-else to pass the future_covariates (check suggestion in the `fit()`).

Also, it would be great to at least check that there is the same number of components in the forecasted series as models in the `self_trained_models` attribute.

Reply:
Hi, I tried the approach with if else for passing the covariates, but it was throwing errors for models which didn't support future covariates

Reply:
Because they were trained without the `future_covariates` (and hence theoretically support them but not in this context, at prediction time?) or because they actually did not support them?

Reply:
Because they actually did not support them, some of the LocalForecastingModels don't support future covariates, so an error is thrown about `future_covariates` not being a parameter for fit() and predict()